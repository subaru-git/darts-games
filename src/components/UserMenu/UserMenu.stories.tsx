import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import UserMenu from './UserMenu';

export default {
  title: 'Components/UserMenu',
  component: UserMenu,
} as ComponentMeta<typeof UserMenu>;

const Template: ComponentStory<typeof UserMenu> = (args) => <UserMenu {...args} />;

export const Default = Template.bind({});
Default.args = {};
