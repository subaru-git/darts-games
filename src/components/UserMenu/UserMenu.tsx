import React, { FC } from 'react';
import {
  Avatar,
  Button,
  Flex,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverCloseButton,
  PopoverContent,
  PopoverHeader,
  PopoverTrigger,
} from '@chakra-ui/react';
import { GoogleLoginButton } from 'react-social-login-buttons';
import { useAuth } from '@/contexts/AuthContext';
import { GoogleLogin, Logout } from '@/lib/Auth';

const UserMenu: FC = () => {
  const user = useAuth();
  return (
    <>
      {user ? (
        <Menu>
          <MenuButton>
            <Avatar name={user.name} src={user.photoURL} size={'sm'} />
          </MenuButton>
          <MenuList>
            <MenuItem onClick={Logout}>Logout</MenuItem>
          </MenuList>
        </Menu>
      ) : (
        <Popover>
          <PopoverTrigger>
            <Button colorScheme={'blue'} size={{ base: 'sm', md: 'md' }}>
              Login
            </Button>
          </PopoverTrigger>
          <PopoverContent>
            <PopoverHeader border={0} fontWeight='semibold'>
              Login
            </PopoverHeader>
            <PopoverArrow />
            <PopoverCloseButton />
            <PopoverBody>
              <Flex justifyContent={'center'}>
                <GoogleLoginButton onClick={GoogleLogin} aria-label={'login with google'} />
              </Flex>
            </PopoverBody>
          </PopoverContent>
        </Popover>
      )}
    </>
  );
};

export default UserMenu;
