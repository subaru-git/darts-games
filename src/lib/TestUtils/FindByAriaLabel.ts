export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
