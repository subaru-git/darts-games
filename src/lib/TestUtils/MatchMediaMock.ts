import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
