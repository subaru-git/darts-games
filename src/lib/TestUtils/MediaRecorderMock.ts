export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
