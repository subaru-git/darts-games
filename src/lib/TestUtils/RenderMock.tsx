import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
