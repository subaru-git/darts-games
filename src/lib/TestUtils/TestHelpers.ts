import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}