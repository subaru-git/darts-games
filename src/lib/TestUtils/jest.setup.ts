import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
