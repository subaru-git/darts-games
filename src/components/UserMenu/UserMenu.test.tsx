import UserMenu from './UserMenu';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<UserMenu />);
  expect(container).toMatchSnapshot();
});
