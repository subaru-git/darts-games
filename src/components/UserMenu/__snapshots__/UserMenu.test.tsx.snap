// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <button
    aria-controls="popover-content-:r1:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="chakra-button css-1pnao28"
    id="popover-trigger-:r1:"
    type="button"
  >
    Login
  </button>
  <div
    class="chakra-popover__popper css-1qq679y"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <section
      aria-describedby="popover-body-:r1:"
      aria-labelledby="popover-header-:r1:"
      class="chakra-popover__content css-sjj62m"
      id="popover-content-:r1:"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
      tabindex="-1"
    >
      <header
        class="chakra-popover__header css-gdvycv"
        id="popover-header-:r1:"
      >
        Login
      </header>
      <div
        class="chakra-popover__arrow-positioner css-0"
        data-popper-arrow=""
        style="position: absolute;"
      >
        <div
          class="chakra-popover__arrow css-0"
          data-popper-arrow-inner=""
        />
      </div>
      <button
        aria-label="Close"
        class="chakra-popover__close-btn css-1o8qips"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="chakra-popover__body css-1ews2c8"
        id="popover-body-:r1:"
      >
        <div
          class="css-1l4w6pd"
        >
          <button
            style="display: block; border: 0px; border-radius: 3px; box-shadow: rgba(0, 0, 0, 0.5) 0 1px 2px; color: black; cursor: pointer; font-size: 19px; margin: 5px; width: calc(100% - 10px); overflow: hidden; padding: 0px 10px; user-select: none; height: 50px; background: white;"
            type="button"
          >
            <div
              style="align-items: center; display: flex; height: 100%;"
            >
              <div
                style="display: flex; justify-content: center; min-width: 26px;"
              >
                <svg
                  height="26px"
                  viewBox="0 0 64 64"
                  width="26px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <lineargradient
                    gradientTransform="matrix(1 0 0 -1 0 66)"
                    gradientUnits="userSpaceOnUse"
                    id="95yY7w43Oj6n2vH63j6HJb"
                    x1="29.401"
                    x2="29.401"
                    y1="4.064"
                    y2="106.734"
                  >
                    <stop
                      offset="0"
                      stop-color="#ff5840"
                    />
                    <stop
                      offset=".007"
                      stop-color="#ff5840"
                    />
                    <stop
                      offset=".989"
                      stop-color="#fa528c"
                    />
                    <stop
                      offset="1"
                      stop-color="#fa528c"
                    />
                  </lineargradient>
                  <path
                    d="M47.46,15.5l-1.37,1.48c-1.34,1.44-3.5,1.67-5.15,0.6c-2.71-1.75-6.43-3.13-11-2.37	c-4.94,0.83-9.17,3.85-11.64, 7.97l-8.03-6.08C14.99,9.82,23.2,5,32.5,5c5,0,9.94,1.56,14.27,4.46	C48.81,10.83,49.13,13.71,47.46,15.5z"
                    fill="url(#95yY7w43Oj6n2vH63j6HJb)"
                  />
                  <lineargradient
                    gradientTransform="matrix(1 0 0 -1 0 66)"
                    gradientUnits="userSpaceOnUse"
                    id="95yY7w43Oj6n2vH63j6HJc"
                    x1="12.148"
                    x2="12.148"
                    y1=".872"
                    y2="47.812"
                  >
                    <stop
                      offset="0"
                      stop-color="#feaa53"
                    />
                    <stop
                      offset=".612"
                      stop-color="#ffcd49"
                    />
                    <stop
                      offset="1"
                      stop-color="#ffde44"
                    />
                  </lineargradient>
                  <path
                    d="M16.01,30.91c-0.09,2.47,0.37,4.83,1.27,6.96l-8.21,6.05c-1.35-2.51-2.3-5.28-2.75-8.22	c-1.06-6.88,0.54-13.38, 3.95-18.6l8.03,6.08C16.93,25.47,16.1,28.11,16.01,30.91z"
                    fill="url(#95yY7w43Oj6n2vH63j6HJc)"
                  />
                  <lineargradient
                    gradientTransform="matrix(1 0 0 -1 0 66)"
                    gradientUnits="userSpaceOnUse"
                    id="95yY7w43Oj6n2vH63j6HJd"
                    x1="29.76"
                    x2="29.76"
                    y1="32.149"
                    y2="-6.939"
                  >
                    <stop
                      offset="0"
                      stop-color="#42d778"
                    />
                    <stop
                      offset=".428"
                      stop-color="#3dca76"
                    />
                    <stop
                      offset="1"
                      stop-color="#34b171"
                    />
                  </lineargradient>
                  <path
                    d="M50.45,51.28c-4.55,4.07-10.61,6.57-17.36,6.71C22.91,58.2,13.66,52.53,9.07,43.92l8.21-6.05	C19.78,43.81, 25.67,48,32.5,48c3.94,0,7.52-1.28,10.33-3.44L50.45,51.28z"
                    fill="url(#95yY7w43Oj6n2vH63j6HJd)"
                  />
                  <lineargradient
                    gradientTransform="matrix(1 0 0 -1 0 66)"
                    gradientUnits="userSpaceOnUse"
                    id="95yY7w43Oj6n2vH63j6HJe"
                    x1="46"
                    x2="46"
                    y1="3.638"
                    y2="35.593"
                  >
                    <stop
                      offset="0"
                      stop-color="#155cde"
                    />
                    <stop
                      offset=".278"
                      stop-color="#1f7fe5"
                    />
                    <stop
                      offset=".569"
                      stop-color="#279ceb"
                    />
                    <stop
                      offset=".82"
                      stop-color="#2cafef"
                    />
                    <stop
                      offset="1"
                      stop-color="#2eb5f0"
                    />
                  </lineargradient>
                  <path
                    d="M59,31.97c0.01,7.73-3.26,14.58-8.55,19.31l-7.62-6.72c2.1-1.61,3.77-3.71,4.84-6.15 c0.29-0.66-0.2-1.41-0.92-1.41H37c-2.21,0-4-1.79-4-4v-2c0-2.21,1.79-4,4-4h17C56.75,27,59,29.22,59,31.97z"
                    fill="url(#95yY7w43Oj6n2vH63j6HJe)"
                  />
                </svg>
              </div>
              <div
                style="width: 10px;"
              />
              <div
                style="text-align: left; width: 100%;"
              >
                Log in with Google
              </div>
            </div>
          </button>
        </div>
      </div>
    </section>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
